### PR TITLE
implement postpaid services

### PR DESCRIPTION
**Pull request by kelvin888**
Implement postpaid services
1. Pay bill
2. Download bill statement
3. Buy bundle

### CODE REVIEW

File: src/components/RadioInput/index.tsx [hunk 1]
```diff
@@ -10,13 +10,10 @@ export const RadioInput: React.FC<{
...
 }> = (props) => {
   const { children, checked = false, onChange = () => null, ...rest } = props;
-  const [state, setState] = useState(checked);
 
   const handleClick = () => {
-    setState(!state);
-
     const event = {
-      target: { value: !state },
+      target: { value: checked },
```
Comment (henperi) [inline src/components/RadioInput/index.tsx]: I feel like these changes may break other parts of the app, did you test this change?

It seems like the checked state will never be toggled, except you maintain state from outside the component

File: src/components/RadioInput/index.tsx [hunk 2]
```diff
@@ -10,13 +10,10 @@ export const RadioInput: React.FC<{
   onChange?: (event: RadioInputEvent) => void;
 }> = (props) => {
   const { children, checked = false, onChange = () => null, ...rest } = props;
-  const [state, setState] = useState(checked);
```
Comment (henperi) [inline src/components/RadioInput/index.tsx]: This allows the component to maintain it's own state. Why do you think removing it is a better approach?

File: src/components/RadioInput/useRadioInput.tsx [hunk 3]
```diff
@@ -38,4 +41,59 @@ function useRadioInput(defaultChecked?: boolean) {
...
+ * @returns an object containg a RadioInput component, checked and setChecked
+ */
+export function useRadioGroup() {
+  const values = [
+    { value: '09081013623', label: '09081013623', checked: true },
+    { value: '09081013623', label: '09081013623', checked: true },
+    { value: '09081013623', label: '09081013623', checked: false },
+    { value: '09081013623', label: '09081013623', checked: false },
+    { value: '09081013623', label: '09081013623', checked: false },
+    { value: '09081013623', label: '09081013623', checked: false },
```
Comment (henperi) [inline src/components/RadioInput/useRadioInput.tsx]: We only have static values here.

File: src/pages/Postpaid/PayBill.tsx [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+import { Modal } from '../../components/Modal';
+import { SuccessResp } from './Interface';
+import { useLazyFetch, usePost } from '../../customHooks/useRequests';
+import { logger } from '../../utils/logger';
+import { ErrorBox } from '../../components/ErrorBox';
+import { useGlobalStore } from '../../store';
+import { Row } from '../../components/Row';
+import { Spinner } from '../../components/Spinner';
+
+interface CorporateDetailsResp {
```
Comment (henperi) [inline src/pages/Postpaid/PayBill.tsx]: You defined this interface in `src/pages/Postpaid/Interface.ts ` consider importing it instead

**Review by henperi — CHANGES_REQUESTED** [review]
Just a few comments, else every other thing looks really good

File: src/components/RadioInput/useRadioInput.tsx (hunk 3 above)
Comment (kelvin888) [reply src/components/RadioInput/useRadioInput.tsx]: I discontinued this. i'll take it out

File: src/pages/Postpaid/PayBill.tsx (hunk 4 above)
Comment (kelvin888) [reply src/pages/Postpaid/PayBill.tsx]: I'll check this and revert

**Pull request merged**